# Upstream: https://bugs.webkit.org/show_bug.cgi?id=123480
Index: webkitgtk-2.4.9/Source/WebCore/platform/gtk/WidgetBackingStoreGtkX11.cpp
===================================================================
--- webkitgtk-2.4.9.orig/Source/WebCore/platform/gtk/WidgetBackingStoreGtkX11.cpp
+++ webkitgtk-2.4.9/Source/WebCore/platform/gtk/WidgetBackingStoreGtkX11.cpp
@@ -30,6 +30,11 @@
 
 namespace WebCore {
 
+static void free_pixmap (void *data)
+{
+  XFreePixmap(GDK_SCREEN_XDISPLAY(gdk_screen_get_default()), GPOINTER_TO_UINT(data));
+}
+
 PassOwnPtr<WidgetBackingStore> WidgetBackingStoreGtkX11::create(GtkWidget* widget, const IntSize& size, float deviceScaleFactor)
 {
     return adoptPtr(new WidgetBackingStoreGtkX11(widget, size, deviceScaleFactor));
@@ -52,13 +57,14 @@ WidgetBackingStoreGtkX11::WidgetBackingS
         GDK_VISUAL_XVISUAL(visual), scaledSize.width(), scaledSize.height()));
 
     cairoSurfaceSetDeviceScale(m_surface.get(), deviceScaleFactor, deviceScaleFactor);
+    static cairo_user_data_key_t key;
+    cairo_surface_set_user_data(cairoSurface(), &key, GUINT_TO_POINTER(m_pixmap), (cairo_destroy_func_t)free_pixmap);
 }
 
 WidgetBackingStoreGtkX11::~WidgetBackingStoreGtkX11()
 {
     // The pixmap needs to exist when the surface is destroyed, so begin by clearing it.
     m_surface.clear();
-    XFreePixmap(m_display, m_pixmap);
     XFreeGC(m_display, m_gc);
 }
 
