Index: graphviz-2.38.0/cmd/dot/Makefile.am
===================================================================
--- graphviz-2.38.0.orig/cmd/dot/Makefile.am
+++ graphviz-2.38.0/cmd/dot/Makefile.am
@@ -48,7 +48,8 @@ dot_SOURCES = dot.c no_builtins.c
 dot_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
 dot_LDADD = \
 	$(top_builddir)/lib/gvc/libgvc.la \
-	$(top_builddir)/lib/cgraph/libcgraph.la
+	$(top_builddir)/lib/cgraph/libcgraph.la \
+	$(MATH_LIBS)
 
 install-data-hook:
 	(cd $(DESTDIR)$(man1dir); for i in $(linkedman); do rm -f $$i; $(LN_S) dot.1 $$i; done;)
