From 6d9c2eafa583507b1597f9c1001a95692d970680 Mon Sep 17 00:00:00 2001
From: John Ellson <ellson@research.att.com>
Date: Fri, 23 Jan 2015 11:50:15 -0500
Subject: [PATCH] apply Redhat's graphviz-2.38.0-rbconfig.patch

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f5e74d3..0e41f7a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1593,7 +1593,7 @@ else
 	    if test "x$DARWIN" = "xyes"; then
 	      RUBY_CFLAGS=`echo $RUBY_CFLAGS | sed 's/powerpc/universal/'`
 	    fi
-            RUBY_LIBS="-L`$RUBY $srcdir/config/config_ruby.rb lib` `$RUBY -rrbconfig -e \"puts Config::CONFIG[['LIBRUBYARG_SHARED']]\"`"
+            RUBY_LIBS="-L`$RUBY $srcdir/config/config_ruby.rb lib` `$RUBY -rrbconfig -e \"puts RbConfig::CONFIG[['LIBRUBYARG_SHARED']]\"`"
             RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb vendorarchdir`"
 	    if test "x$RUBY_INSTALL_DIR" = "x"; then
 	  	  RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb sitearchdir`"
