Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -106,11 +106,7 @@ AC_ARG_VAR(LIBPOSTFIX, [subscript to lib
 if test -z "$LIBPOSTFIX"; then
   case "${host_os}" in
     *linux* )
-      case "${host_cpu}" in
-        aarch64 | powerpc64 | powerpc64le | s390x | x86_64 | sparc64 )
-          LIBPOSTFIX="64"
-          ;;
-      esac
+      # nothing to do for multiarch systems
       ;;
     *solaris* )
       case "${host_cpu}" in
@@ -1655,10 +1651,14 @@ fi
 if test "x$use_tcl" = "x"; then
   # Some systems don't have Tcl.  Don't build
   # the Tcl products if we don't have the library.
-  if test -f ${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tcl${TCL_VERSION_FOUND}/tclConfig.sh; then
+  if test -f ${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tcl${TCL_VERSION_FOUND}/tclConfig.sh; then
+    TCLCONFIG=${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tcl${TCL_VERSION_FOUND}/tclConfig.sh
+  elif test -f ${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tcl${TCL_VERSION_FOUND}/tclConfig.sh; then
     TCLCONFIG=${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tcl${TCL_VERSION_FOUND}/tclConfig.sh
   else
-    if test -f ${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tclConfig.sh; then
+    if test -f ${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tclConfig.sh; then
+      TCLCONFIG=${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tclConfig.sh
+    elif test -f ${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tclConfig.sh; then
       TCLCONFIG=${TCLSH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tclConfig.sh
     else
       if test -f ${TCLSH_EXEC_PREFIX}/lib/tclConfig.sh; then
@@ -1803,10 +1803,14 @@ if test "x$use_tk" = "x"; then
     # Some systems don't have Tcl.  Don't build
     # Some systems have Tcl, but not Tk.  Don't build
     # the Tk products if we don't have the library.
-    if test -f ${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tk${TK_VERSION_FOUND}/tkConfig.sh; then
+    if test -f ${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tk${TK_VERSION_FOUND}/tkConfig.sh; then
+        TKCONFIG=${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tk${TK_VERSION_FOUND}/tkConfig.sh
+    elif test -f ${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tk${TK_VERSION_FOUND}/tkConfig.sh; then
         TKCONFIG=${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tk${TK_VERSION_FOUND}/tkConfig.sh
     else
-        if test -f ${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tkConfig.sh; then
+        if test -f ${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tkConfig.sh; then
+	    TKCONFIG=${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/${DEB_HOST_MULTIARCH}/tkConfig.sh
+        elif test -f ${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tkConfig.sh; then
 	    TKCONFIG=${WISH_EXEC_PREFIX}/lib${LIBPOSTFIX}/tkConfig.sh
 	else
             if test -f ${WISH_EXEC_PREFIX}/lib/tkConfig.sh; then
