Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -1566,7 +1566,7 @@ else
           RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb vendorarchdir`"
 	fi
       ],[
-        PKG_CHECK_MODULES([RUBY], [ruby-1.9],[
+        PKG_CHECK_MODULES([RUBY], [ruby-$(ruby -e 'puts RUBY_VERSION.split(".").first(2).join(".")')],[
 	  AC_CHECK_PROG(RUBY,ruby,ruby)
           if test "x$RUBY" = "x"; then
             use_ruby="No (ruby not available)"
