Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -54,7 +54,9 @@ GVPLUGIN_REVISION=0
 # precedence over adding, so set to 0 if both happened.
 GVPLUGIN_AGE=0
 
-GVPLUGIN_CONFIG_FILE="config$GVPLUGIN_CURRENT"
+# upstream didn't bump GVPLUGIN_CURRENT between 2.26 (gvc5) and 2.38 (gvc6).
+# just append something to differentiate this, don't go ahead before upstream
+GVPLUGIN_CONFIG_FILE="config$GVPLUGIN_CURRENT"a
 AC_DEFINE_UNQUOTED(GVPLUGIN_CONFIG_FILE,"$GVPLUGIN_CONFIG_FILE",[Filename for plugin configuration file.])
 AC_SUBST([GVPLUGIN_CONFIG_FILE])
 
