diff -ur phodav-2.3/libphodav/phodav-method-proppatch.c phodav-2.3.new/libphodav/phodav-method-proppatch.c
--- phodav-2.3/libphodav/phodav-method-proppatch.c	2019-02-26 15:12:17.000000000 -0300
+++ phodav-2.3.new/libphodav/phodav-method-proppatch.c	2021-07-16 23:43:54.276769095 -0300
@@ -22,7 +22,7 @@
 
 #include <sys/types.h>
 #ifdef HAVE_ATTR_XATTR_H
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #endif
 
 
diff -ur phodav-2.3/meson.build phodav-2.3.new/meson.build
--- phodav-2.3/meson.build	2019-02-26 15:12:17.000000000 -0300
+++ phodav-2.3.new/meson.build	2021-07-16 23:43:25.438767159 -0300
@@ -53,7 +53,7 @@
   )
 endif
 
-udev = dependency('udev', required : get_option('systemd'))
+udev = dependency('udev', required : get_option('udev'))
 if udev.found()
   install_data(
     'data/70-spice-webdavd.rules',
@@ -61,7 +61,7 @@
   )
 endif
 
-if compiler.has_header('attr/xattr.h')
+if compiler.has_header('sys/xattr.h')
   conf.set('HAVE_ATTR_XATTR_H', 1)
 endif
 
diff -ur phodav-2.3/meson_options.txt phodav-2.3.new/meson_options.txt
--- phodav-2.3/meson_options.txt	2019-02-26 15:12:17.000000000 -0300
+++ phodav-2.3.new/meson_options.txt	2021-07-16 23:43:25.438767159 -0300
@@ -10,6 +10,10 @@
        type : 'feature',
        description : 'Install systemd units/configuration')
 
+option('udev',
+       type : 'feature',
+       description : 'Install udev rules/configuration')
+
 option('build-id',
        type : 'integer',
        value : 0,
