Subject: Don't override DefaultStandardOutput.
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=741124
Forwarded: no
---
 data/accounts-daemon.service.in |    1 -
 1 file changed, 1 deletion(-)

--- accountsservice.orig/data/accounts-daemon.service.in
+++ accountsservice/data/accounts-daemon.service.in
@@ -11,7 +11,6 @@ Wants=nss-user-lookup.target
 Type=dbus
 BusName=org.freedesktop.Accounts
 ExecStart=@libexecdir@/accounts-daemon
-StandardOutput=syslog
 
 [Install]
 # We pull this in by graphical.target instead of waiting for the bus
