From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Fri, 20 May 2016 14:31:59 +0800
Subject: lower_dbusmenu_requirement

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1b09f7d..bf9cc03 100644
--- a/configure.ac
+++ b/configure.ac
@@ -104,7 +104,7 @@ GTK_CFLAGS="$GTK_CFLAGS -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_4"
 AC_ARG_WITH([appindicator], AS_HELP_STRING([--with-appindicator|--without-appindicator], [Build with libappindicator support instead of xembed systray support.]))
 if test "$with_appindicator" == "yes"; then
 	PKG_CHECK_MODULES(APPINDICATOR, appindicator3-0.1)
-	PKG_CHECK_MODULES(DBUSMENU, dbusmenu-gtk3-0.4 >= 16.04.0)
+	PKG_CHECK_MODULES(DBUSMENU, dbusmenu-gtk3-0.4 >= 12.10)
 	AC_DEFINE([WITH_APPINDICATOR], 1, [Enable using libappindicator])
 fi
 
