From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Fri, 20 May 2016 14:31:26 +0800
Subject: reorder-indicator-items

---
 src/applet.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/applet.c b/src/applet.c
index 8406739..b566709 100644
--- a/src/applet.c
+++ b/src/applet.c
@@ -1807,15 +1807,15 @@ applet_connection_info_cb (NMApplet *applet)
  * Generate the contextual popup menu.
  *
  */
-static GtkWidget *nma_context_menu_create (NMApplet *applet)
+static GtkWidget *nma_context_menu_create (NMApplet *applet, GtkMenuShell *menu)
 {
-	GtkMenuShell *menu;
 	guint id;
 	static gboolean icons_shown = FALSE;
 
 	g_return_val_if_fail (applet != NULL, NULL);
 
-	menu = GTK_MENU_SHELL (gtk_menu_new ());
+	if (!menu)
+		menu = GTK_MENU_SHELL (gtk_menu_new ());
 
 	if (G_UNLIKELY (icons_shown == FALSE)) {
 		GtkSettings *settings = gtk_widget_get_settings (GTK_WIDGET (menu));
@@ -2005,9 +2005,10 @@ applet_update_indicator_menu (gpointer user_data)
 #ifdef WITH_APPINDICATOR
 	GtkWidget *menu;
 
-	menu = nma_context_menu_create (applet);
+	menu = gtk_menu_new ();
 	nma_menu_show_cb (menu, applet);
 	nma_menu_add_separator_item (menu);
+	menu = nma_context_menu_create (applet, GTK_MENU_SHELL (menu));
 	nma_context_menu_update (applet);
 
 	app_indicator_set_menu (applet->app_indicator, GTK_MENU (menu));
@@ -3247,7 +3248,7 @@ setup_widgets (NMApplet *applet)
 		g_signal_connect (applet->status_icon, "popup-menu",
 				  G_CALLBACK (status_icon_popup_menu_cb), applet);
 
-		applet->context_menu = nma_context_menu_create (applet);
+		applet->context_menu = nma_context_menu_create (applet, NULL);
 		if (!applet->context_menu)
 			return FALSE;
 	}
