Description: fix compilation with --no-add-needed
Origin: http://bazaar.launchpad.net/~ubuntu-branches/ubuntu/quantal/libpst/quantal/view/head:/debian/patches/06-ld-no-add-needed.patch

Index: repo/src/Makefile.am
===================================================================
--- repo.orig/src/Makefile.am	2013-01-16 11:18:02.946848930 +0100
+++ repo/src/Makefile.am	2013-01-16 11:18:02.942848959 +0100
@@ -80,7 +80,7 @@
 endif
 
 libpst_la_SOURCES     = $(common_source) $(common_header)
-libpst_la_LIBADD      = $(LTLIBICONV)
+libpst_la_LIBADD      = $(LTLIBICONV) -lpthread
 
 EXTRA_DIST =
 if !NEED_XGETOPT
@@ -92,7 +92,7 @@
 
 # the library search path.
 lspst_LDADD       = $(all_libraries) $(PSTLIB) $(LTLIBICONV)
-readpst_LDADD     = $(all_libraries) $(PSTLIB) $(LTLIBICONV) $(REGEXLIB) $(GSF_LIBS)
+readpst_LDADD     = $(all_libraries) $(PSTLIB) $(LTLIBICONV) $(REGEXLIB) $(GSF_LIBS) -lpthread
 pst2ldif_LDADD    = $(all_libraries) $(PSTLIB) $(LTLIBICONV)
 pst2dii_LDADD     = $(all_libraries) $(PSTLIB) $(LTLIBICONV) -lgd
 deltasearch_LDADD = $(all_libraries) $(PSTLIB) $(LTLIBICONV)
