diff -Nur gdk-pixbuf-2.22.0.orig/gdk-pixbuf/Makefile.in gdk-pixbuf-2.22.0/gdk-pixbuf/Makefile.in
--- gdk-pixbuf-2.22.0.orig/gdk-pixbuf/Makefile.in	2010-09-21 13:28:57.000000000 -0500
+++ gdk-pixbuf-2.22.0/gdk-pixbuf/Makefile.in	2010-10-09 10:24:16.917235308 -0500
@@ -795,7 +795,7 @@
 LIBJASPER = @LIBJASPER@
 LIBJPEG = @LIBJPEG@
 LIBOBJS = @LIBOBJS@
-LIBPNG = @LIBPNG@
+LIBPNG = @LIBPNG@ -lz
 LIBS = @LIBS@
 LIBTIFF = @LIBTIFF@
 LIBTOOL = @LIBTOOL@
