Index: libgweather-3.18.1/build-aux/ltmain.sh
===================================================================
--- libgweather-3.18.1.orig/build-aux/ltmain.sh	2015-10-12 18:27:58.347107854 +0200
+++ libgweather-3.18.1/build-aux/ltmain.sh	2015-10-12 18:27:58.343107771 +0200
@@ -7220,6 +7220,11 @@
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+      
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -7604,6 +7609,15 @@
 	lib=
 	found=false
 	case $deplib in
+	-Wl,--as-needed)
+	  if test "$linkmode,$pass" = "prog,link"; then
+	    compile_deplibs="$deplib $compile_deplibs"
+	    finalize_deplibs="$deplib $finalize_deplibs"
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
         |-threads|-fopenmp|-openmp|-mp|-xopenmp|-omp|-qsmp=*)
 	  if test prog,link = "$linkmode,$pass"; then
