Description: Use the stable libsecret.
Author: Robert Ancell <robert.ancell@canonical.com>

Index: seahorse-3.10.1/configure.ac
===================================================================
--- seahorse-3.10.1.orig/configure.ac	2013-12-16 17:43:18.815496522 +1300
+++ seahorse-3.10.1/configure.ac	2013-12-16 17:43:18.811496523 +1300
@@ -339,7 +339,7 @@
 # -----------------------------------------------------------------------------
 # LIBSECRET
 
-PKG_CHECK_MODULES([LIBSECRET], libsecret-unstable >= $LIBSECRET_REQUIRED,
+PKG_CHECK_MODULES([LIBSECRET], libsecret-1 >= $LIBSECRET_REQUIRED,
                   [enable_gnome_keyring=yes],[enable_gnome_keyring=no])
 
 if test "$enable_gnome_keyring" = "no"; then
