Description: Let help buttons point to valid documentation references.
Bug: https://bugzilla.gnome.org/659834
Bug-Ubuntu: https://launchpad.net/bugs/853137
Author: Sebastien Bacher <seb128@ubuntu.com>

Index: seahorse-3.8.2/libseahorse/seahorse-widget.c
===================================================================
--- seahorse-3.8.2.orig/libseahorse/seahorse-widget.c	2013-05-21 13:27:59.170407441 +0200
+++ seahorse-3.8.2/libseahorse/seahorse-widget.c	2013-05-21 13:27:59.166407441 +0200
@@ -450,6 +450,9 @@
     if (g_str_equal (swidget->name, "key-manager") || 
         g_str_equal (swidget->name, "keyserver-results")) {
         document = g_strdup ("help:" PACKAGE "/introduction");
+    } else if (g_str_equal (swidget->name, "gkr-keyring") || 
+        g_str_equal (swidget->name, "gkr-item-properties")) {
+        document = g_strdup ("help:" PACKAGE "/keyring");
     } else {
         document = g_strdup_printf ("help:" PACKAGE "/%s", swidget->name);
     }
