# DP: Relax the mpdecimal version check

Index: b/Modules/_decimal/_decimal.c
===================================================================
--- a/Modules/_decimal/_decimal.c
+++ b/Modules/_decimal/_decimal.c
@@ -39,10 +39,6 @@
 #include "memory.h"
 
 
-#if !defined(MPD_VERSION_HEX) || MPD_VERSION_HEX < 0x02040100
-  #error "libmpdec version >= 2.4.1 required"
-#endif
-
 
 /*
  * Type sizes with assertions in mpdecimal.h and pyport.h:
